test/evp_extra_test.c

```diff
@@ -2512,6 +2573,7 @@ int setup_tests(void)
     ADD_TEST(test_rand_agglomeration);
     ADD_ALL_TESTS(test_evp_iv, 10);
     ADD_TEST(test_EVP_rsa_pss_with_keygen_bits);
+    ADD_ALL_TESTS(test_ecpub, OSSL_NELEM(ecpub_nids));
```

Comment:
This, as well as `test_ecpub`, needs to be wrapped in a `#ifndef OPENSSL_NO_EC` guard.  From the CI builds:

```
In file included from test/evp_extra_test.c:33:0:
test/evp_extra_test.c: In function 'test_ecpub':
test/evp_extra_test.c:2456:32: error: implicit declaration of function 'EC_KEY_new_by_curve_name'; did you mean 'EVP_PKEY_get_group_name'? [-Werror=implicit-function-declaration]
             || !TEST_ptr((ec = EC_KEY_new_by_curve_name(nid)))
                                ^
test/testutil.h:449:64: note: in definition of macro 'TEST_ptr'
 # define TEST_ptr(a)          test_ptr(__FILE__, __LINE__, #a, a)
                                                                ^
test/evp_extra_test.c:2456:30: error: assignment makes pointer from integer without a cast [-Werror=int-conversion]
             || !TEST_ptr((ec = EC_KEY_new_by_curve_name(nid)))
                              ^
test/testutil.h:449:64: note: in definition of macro 'TEST_ptr'
 # define TEST_ptr(a)          test_ptr(__FILE__, __LINE__, #a, a)
                                                                ^
test/evp_extra_test.c:2457:27: error: implicit declaration of function 'EVP_PKEY_assign_EC_KEY'; did you mean 'EVP_PKEY_assign_RSA'? [-Werror=implicit-function-declaration]
             || !TEST_true(EVP_PKEY_assign_EC_KEY(pkeytmp, ec)))
Makefile:11327: recipe for target 'test/evp_extra_test-bin-evp_extra_test.o' failed
                           ^
test/testutil.h:462:66: note: in definition of macro 'TEST_true'
 # define TEST_true(a)         test_true(__FILE__, __LINE__, #a, (a) != 0)
                                                                  ^
test/evp_extra_test.c:2475:5: error: implicit declaration of function 'EC_KEY_free'; did you mean 'EVP_KEM_free'? [-Werror=implicit-function-declaration]
     EC_KEY_free(ec);
     ^~~~~~~~~~~
     EVP_KEM_free
cc1: all warnings being treated as errors
```

Reply:
Thanks for spotting that; I will update shortly.

But while I'm happy to have review on the tests, I was hoping more to get your insight on the library code that is being tested (and found wanting)...